spec/support/shared_examples.rb

```diff
@@ -0,0 +1,14 @@
+RSpec.shared_examples '#canonicalize' do |provider|
```

Comment:
If you check out https://github.com/puppetlabs/cisco_ios/blob/00e6a9a6f9bd51c1cca280d67eba1f278d6122e9/spec/spec_helper_local.rb#L49 , you don't need to pass in the `described_class`. it'll be available through rspec's magic scoping trix.

Then you can also use the `it_behaves_like` syntax and give this a nicer name, e.g. `it_behaves_like 'a noop canonicalizer'`